File: core/src/main/java/ru/pocketbyte/locolaser/config/source/EmptySourceConfig.java
package ru.pocketbyte.locolaser.config.source;

import ru.pocketbyte.locolaser.resource.PlatformResources;

import java.util.HashSet;
import java.util.Set;

public class EmptySourceConfig implements SourceConfig {

    public static final String TYPE = "null";

    private Set<String> mLocales;

    public EmptySourceConfig() {
        mLocales = new HashSet<>();
        mLocales.add(PlatformResources.BASE_LOCALE);
    }

    @Override
    public String getType() {
        return TYPE;
    }

    @Override
    public Source open() {
        return new EmptySource(this);
    }

    @Override
    public Set<String> getLocales() {
        return mLocales;
    }
}

File: core/src/main/java/ru/pocketbyte/locolaser/config/source/EmptySourceConfig.kt
package ru.pocketbyte.locolaser.config.source

import ru.pocketbyte.locolaser.resource.PlatformResources

class EmptySourceConfig : SourceConfig {

    companion object {
        const val TYPE = "null"
    }

    override val type: String = TYPE
    override val locales: Set<String> = setOf(PlatformResources.BASE_LOCALE)

    override fun open(): Source? {
        return EmptySource()
    }
}
